packages/mui-material/src/BottomNavigationAction/BottomNavigationAction.test.js

```diff
@@ -73,4 +73,170 @@ describe('<BottomNavigationAction />', () => {
+
+      expect(handleClick.callCount).to.equal(1);
+    });
+
+    it('should not fire onClick twice on touch tap', () => {
+      const handleClick = spy();
+
+      // Need disableTouchRipple to avoid ripple missing act (async setState after touchEnd)
+      const { getByRole, container } = render(
+        <BottomNavigationAction onClick={handleClick} disableTouchRipple />,
+      );
+
+      fireEvent.touchStart(container.firstChild, {
+        touches: [
+          new Touch({
+            identifier: 1,
+            target: container,
+            clientX: 42,
+            clientY: 42,
+          }),
+        ],
+      });
+
+      fireEvent.touchEnd(container.firstChild, {
+        changedTouches: [
+          new Touch({
+            identifier: 1,
+            target: container,
+            clientX: 42,
+            clientY: 42,
+          }),
+        ],
+      });
+
+      getByRole('button').click();
+
+      clock.tick(100);
+
+      expect(handleClick.callCount).to.equal(1);
+    });
+
+    it('should not fire onClick if swiping', () => {
+      const handleClick = spy();
+
+      // Need disableTouchRipple to avoid ripple missing act (async setState after touchEnd)
+      const { container } = render(
+        <BottomNavigationAction onClick={handleClick} disableTouchRipple />,
+      );
+
+      fireEvent.touchStart(container.firstChild, {
+        touches: [
+          new Touch({
+            identifier: 1,
+            target: container,
+            clientX: 42,
+            clientY: 42,
+          }),
+        ],
+      });
+
+      fireEvent.touchEnd(container.firstChild, {
+        changedTouches: [
+          new Touch({
+            identifier: 1,
+            target: container,
+            clientX: 84,
+            clientY: 84,
+          }),
+        ],
+      });
+
+      clock.tick(10);
+
+      expect(handleClick.callCount).to.equal(0);
+    });
+
+    it('should forward onTouchStart and onTouchEnd events', () => {
+      const handleTouchStart = spy();
+      const handleTouchEnd = spy();
+
+      // Need disableTouchRipple to avoid ripple missing act (async setState after touchEnd).
+      const { container } = render(
+        <BottomNavigationAction
+          onTouchStart={handleTouchStart}
+          onTouchEnd={handleTouchEnd}
+          disableTouchRipple
+        />,
+      );
+
+      fireEvent.touchStart(container.firstChild, {
+        touches: [
+          new Touch({
+            identifier: 1,
+            target: container,
+            clientX: 42,
+            clientY: 42,
+          }),
+        ],
+      });
+
+      expect(handleTouchStart.callCount).to.be.equals(1);
+
+      fireEvent.touchEnd(container.firstChild, {
+        changedTouches: [
+          new Touch({
+            identifier: 1,
+            target: container,
+            clientX: 84,
+            clientY: 84,
+          }),
+        ],
+      });
+
+      expect(handleTouchEnd.callCount).to.be.equals(1);
+    });
+  });
```

Comment:
How did you come up with this event order? A codesandbox to verify that this is actually happening, is important. For example, it's a bit odd that a child is the target of the touch event but the touch itself is the parent.

Reply:
I do not remember, the PR for this was accepted by @oliviertassinari a year ago.
The only new code in this PR, is the code in commit 6082cebd9d2cd60cb84534b5f5deb91d96f1a191.